exonum/src/storage/proof_map_index/node.rs

```diff
@@ -82,12 +84,25 @@ impl BranchNode {
 
 impl CryptoHash for BranchNode {
     fn hash(&self) -> Hash {
-        HashStream::new()
-            .update(self.child_hash(ChildKind::Left).as_ref())
-            .update(self.child_hash(ChildKind::Right).as_ref())
-            .update(self.child_path(ChildKind::Left).compress().as_ref())
-            .update(self.child_path(ChildKind::Right).compress().as_ref())
-            .hash()
+        // It could be worth investigating whether implementing `smallvec::Array` for
+        // a `[u8; 132]` newtype could improve performance.
+        let mut bytes: SmallVec<[u8; 256]> = SmallVec::new();
+        let left_path = self.child_path(ChildKind::Left);
+        let right_path = self.child_path(ChildKind::Right);
+
+        let hashes_end = HASH_SIZE * 2;
+        let right_path_start = hashes_end + left_path.compressed_len();
+        let total_len = right_path_start + right_path.compressed_len();
+        // You can do without `unsafe` by initializing `bytes` to `smallvec![0; len]`;
+        // as far as I can tell, the decrease in performance is negligible.
```

Comment:
So why not go for it then? I'd rather not try to outsmart the compiler by preventively inserting `unsafe`s without profiling results to back them up.